integration-cli/docker_api_logs_test.go

```diff
@@ -5,24 +5,44 @@ import (
 	"fmt"
 	"net/http"
 	"os/exec"
+	"strings"
+	"time"
 
 	"github.com/go-check/check"
 )
 
 func (s *DockerSuite) TestLogsApiWithStdout(c *check.C) {
-	name := "logs_test"
+	name := "log_test"
+	out, _ := dockerCmd(c, "run", "-d", "-t", "busybox", "/bin/sh", "-c", fmt.Sprintf("trap 'echo %s; exit 0' SIGTERM; while true; do sleep 0.1; done", name))
+	id := strings.TrimSpace(out)
+	if err := waitRun(id); err != nil {
+		c.Fatal(err)
+	}
 
-	runCmd := exec.Command(dockerBinary, "run", "-d", "-t", "--name", name, "busybox", "bin/sh", "-c", "sleep 10 && echo "+name)
-	if out, _, err := runCommandWithOutput(runCmd); err != nil {
-		c.Fatal(out, err)
+	type logOut struct {
+		b      []byte
+		status int
+		err    error
 	}
+	chLog := make(chan logOut)
 
-	status, body, err := sockRequest("GET", fmt.Sprintf("/containers/%s/logs?follow=1&stdout=1&timestamps=1", name), nil)
-	c.Assert(status, check.Equals, http.StatusOK)
-	c.Assert(err, check.IsNil)
+	go func() {
+		statusCode, body, err := sockRequest("GET", fmt.Sprintf("/containers/%s/logs?follow=1&stdout=1&timestamps=1", id), nil)
+		chLog <- logOut{body, statusCode, err}
+	}()
+
+	time.Sleep(100 * time.Millisecond)
```

Comment:
Could we have the container print "name" to the screen, over and over. And then wait for it to show up in the logs?